src/driver/mod.rs

```diff
@@ -136,3 +137,21 @@ pub fn nsm_exit(fd: i32) {
         Err(e) => error!("File of descriptor {} failed to close: {}", fd, e),
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::{Request, NSM_REQUEST_MAX_SIZE};
+    #[test]
+    fn test_nsm_encode_request_to_cbor_exceed_max_size_without_panic() {
```

Comment:
This is just here to ensure if anyone in the future decides to try to get clever with the `buf` type, we'll see this `panic`. For instance, you can't provide `&mut [u8]` here, because it won't resize itself: it will hit `nsm_encode_request_to_cbor`'s `expect`